pkg/reconciler/ingress/lister.go

```diff
@@ -63,30 +63,40 @@ func (l *gatewayPodTargetLister) ListProbeTargets(ctx context.Context, ing *v1al
 func (l *gatewayPodTargetLister) getIngressUrls(ing *v1alpha1.Ingress, gatewayIps []string) ([]status.ProbeTarget, error) {
 	ips := sets.NewString(gatewayIps...)
 
+	localTLS := false
+	externalTLS := false
+	for _, t := range ing.Spec.TLS {
+		// Also include empty for reverse compatibility
+		if t.Visibility == "" || t.Visibility == v1alpha1.IngressVisibilityExternalIP {
+			externalTLS = true
+		}
+		if t.Visibility == v1alpha1.IngressVisibilityClusterLocal {
+			localTLS = true
+		}
+	}
+
 	targets := make([]status.ProbeTarget, 0, len(ing.Spec.Rules))
 	for _, rule := range ing.Spec.Rules {
-		var target status.ProbeTarget
-
-		domains := rule.Hosts
-		scheme := "http"
-
-		if rule.Visibility == v1alpha1.IngressVisibilityExternalIP {
-			target = status.ProbeTarget{
-				PodIPs: ips,
-			}
-			if len(ing.Spec.TLS) != 0 {
-				target.PodPort = strconv.Itoa(int(config.HTTPSPortProb))
-				target.URLs = domainsToURL(domains, "https")
-			} else {
-				target.PodPort = strconv.Itoa(int(config.HTTPPortProb))
-				target.URLs = domainsToURL(domains, scheme)
-			}
-		} else {
-			target = status.ProbeTarget{
-				PodIPs:  ips,
-				PodPort: strconv.Itoa(int(config.HTTPPortInternal)),
-				URLs:    domainsToURL(domains, scheme),
-			}
+		target := status.ProbeTarget{
+			PodIPs: ips,
+		}
+
+		switch {
```

Comment:
You made this so much clean and readable :heart_eyes_cat:

Reply:
Thanks 🙏